lib/nostrum/shard/session.ex

```diff
@@ -86,63 +111,63 @@ defmodule Nostrum.Shard.Session do
 
     case from_handle do
       {new_state, reply} ->
-        {:reply, {:binary, reply}, new_state}
+        :ok = :gun.ws_send(state.conn, {:binary, reply})
+        {:noreply, new_state}
 
       new_state ->
-        {:ok, new_state}
+        {:noreply, new_state}
     end
   end
 
+  def handle_info({:gun_ws, _conn, _stream, {:close, errno, reason}}, state) do
+    Logger.info("Shard websocket closed (errno #{errno}, reason #{inspect(reason)}).")
+    {:noreply, state}
+  end
+
+  def handle_info(
+        {:gun_down, _conn, _proto, _reason, _killed_streams, _unprocessed_streams},
+        state
+      ) do
+    {:noreply, state}
+  end
+
+  def handle_info({:gun_up, worker, _proto}, state) do
+    stream = :gun.ws_upgrade(worker, @gateway_qs)
+    await_ws_upgraded(worker, stream)
+    Logger.warn("Reconnected after connection broke.")
```

Comment:
I think periods at the end of log messages are a little weird.

Reply:
I think your emojis in your review comment are weird, and such is the amassment of exclamation marks in your comment. Erlang software runs like a German car, it's this thing you can always rely on, and in order to stand behind that, the log messages in my applications should read like "Sir, I, the Erlang runtime, have once again done something where everything else would tremble in fear, I have reconnected in a message handler and will now continue to serve your application without any issues and proceed by processing my entire message queue without problems." and not like "yup this is 🐐 i think i reconnected". Therefore, I heavily disagree with this suggestion. This also applies to the comment below.